python/pyhail/dataset.py

```diff
@@ -215,6 +241,30 @@ def annotate_global_table(self, input, root, config=None):
     def annotate_samples_expr(self, condition):
         """Annotate samples with expression.
 
+        **Example**
+
+        Compute per-sample GQ statistics for hets:
+
+        >>> (hc.read('data/example.vds')
+        >>>   .annotate_samples_expr('sa.gqHetStats = gs.filter(g => g.isHet).map(g => g.gq).stats()')
+        >>>   .export_samples('data/samples.txt', 'sample = s, het_gq_mean = sa.gqHetStats.mean'))
+
+        Compute the list of genes with a singleton LOF:
+
+        >>> (hc.read('data/example.vds')
+        >>>   .annotate_variants_table('data/consequence.tsv', 'Variant', code='va.consequence = table.Consequence')
+        >>>   .annotate_variants_expr('va.isSingleton = gs.map(g => g.nNonRefAlleles).sum() == 1')
+        >>>   .annotate_samples_expr('sa.LOF_genes = gs.filter(g => va.isSingleton && g.isHet && va.consequence == "LOF").map(g => va.gene).collect()'))
+
+        **Notes**
+
+        ``condition`` is in sample context so the following symbols are in scope:
+
+        - ``s: Sample``
```

Comment:
Add this relative link to Sample object

:ref:\`sample\`

Likewise variant is

:ref:\`variant\`